app/lib/features/activities/providers/notifiers/notifications_list_notifier.dart

```diff
@@ -0,0 +1,128 @@
+import 'package:acter/common/notifications/notifications.dart';
+import 'package:acter/common/utils/utils.dart';
+import 'package:acter/features/activities/util.dart';
+import 'package:acter/features/home/providers/client_providers.dart';
+import 'package:acter/features/settings/providers/settings_providers.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart' as ffi;
+import 'package:flutter/foundation.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:riverpod_infinite_scroll/riverpod_infinite_scroll.dart';
+import 'dart:async';
+
+class Next {
+  final bool isStart;
+  final String? next;
+
+  const Next({this.isStart = false, this.next});
+}
+
+class NotificationListState extends PagedState<Next?, ffi.Notification> {
+  // We can extends [PagedState] to add custom parameters to our state
+
+  const NotificationListState({
+    List<ffi.Notification>? records,
+    String? error,
+    Next? nextPageKey = const Next(isStart: true),
+    List<Next?>? previousPageKeys,
+  }) : super(records: records, error: error, nextPageKey: nextPageKey);
+
+  NotificationListState addNotification(ffi.Notification n) {
+    final r = records ?? [];
+    r.insert(0, n);
+    return copyWith(
+      records: r,
+    );
+  }
+
+  @override
+  NotificationListState copyWith({
```

Comment:
Do you think we could achieve this by ```freezed``` with minimal code?

Reply:
Maybe, I don't know. I don't know `freezed` enough and was just going from other examples provided ...

Reply:
If I understand correctly, you're overriding ```copyWith```. The ```NotificationListState``` can be simplified with freezed like class with generated constructor and ```copyWith``` method.